stan/math/prim/scal/fun/std_binormal_integral.hpp

```diff
@@ -0,0 +1,208 @@
+           - owens_t(z2, a2);
+  }
+  if (fabs(rho) < 1) {
+    int s = rho > 0 ? 1 : -1;
+    double h = -z1;
+    double k = -z2;
+
+    auto f = [&h, &k](const double x) {
+      using std::cos;
+      using std::sin;
+      using stan::math::square;
+      return exp(0.5 * (-h * h - k * k + h * k * 2 * sin(x)) / square(cos(x)));
+    };
+    double L1, error;
+    boost::math::quadrature::tanh_sinh<double> integrator;
+    double accum
+        = s == 1 ? integrator.integrate(f, 0, asin(rho),
+                                        sqrt(stan::math::EPSILON), &error, &L1)
+                 : integrator.integrate(f, asin(rho), 0,
+                                        sqrt(stan::math::EPSILON), &error, &L1);
+    if (exp(log(L1) - log(accum)) > 1.5) {
+      domain_error(function, "the bivariate normal numeric integral condition",
+                   exp(log(L1) - log(accum)), "",
+                   " indicates the integral is poorly conditioned");
+    }
+
+    accum *= s * 0.5 / pi();
+    accum += Phi(z1) * Phi(z2);
+    return accum;
+  }
+  if (rho == 1)
+    return fmin(Phi(z1), Phi(z2));
+  return z2 > -z1 ? Phi(z1) + Phi(z2) - 1 : 0;
+}
+
+/**
+ * The CDF of the standard bivariate normal (binormal) for the specified
+ * variable 1, z1, and variable 2, z2, given the specified correlation, rho.
+ * rho can each be either a scalar or a vector. If z1, z2, or rho are
+ * autodiff types, the function propagates the gradients of the CDF. Reference
+ * for the gradient of the bivariate normal CDF can be found here:
+ *
+ * blogs.sas.com/content/iml/
+ * 2013/09/20/gradient-of-the-bivariate-normal-cumulative-distribution.html
+ *
+ * @param z1 Scalar random variable 1
+ * @param z2 Scalar random variable 2
+ * @param rho Scalar correlation parameter
+ * for the standard bivariate normal distribution.
+ * @return The probability P(Y1 <= y[1], Y2 <= y[2] | rho).
+ * @throw std::domain_error if the rho is not between -1 and 1 or nan.
+ * @tparam T_z1 type of z1
+ * @tparam T_z2 type of z2
+ * @tparam T_rho type of rho
+ */
+
+template <typename T_z1, typename T_z2, typename T_rho>
+typename return_type<T_z1, T_z2, T_rho>::type std_binormal_integral(
+    const T_z1& z1, const T_z2& z2, const T_rho& rho) {
+  typedef typename partials_return_type<T_z1, T_z2, T_rho>::type partials_type;
+  using stan::math::std_binormal_integral;
+  using stan::math::std_normal_lpdf;
+  using stan::math::value_of_rec;
+  using std::asin;
+  using std::exp;
+  using std::log;
+  using std::sqrt;
+
+  const partials_type z1_dbl = value_of(z1);
+  const partials_type z2_dbl = value_of(z2);
+  const partials_type rho_dbl = value_of(rho);
+
+  partials_type cdf = std_binormal_integral(z1_dbl, z2_dbl, rho_dbl);
+  operands_and_partials<T_z1, T_z2, T_rho> ops_partials(z1, z2, rho);
+
+  if (contains_nonconstant_struct<T_z1, T_z2, T_rho>::value) {
+    const partials_type one_minus_rho_sq = (1 + rho_dbl) * (1 - rho_dbl);
```

Comment:
same naming problem

Reply:
What naming problem are you referring to? I don't see any local variables ending with `_`.